File: src/dao/ForwardDao.kt
package cn.dairo.cls.dao

import cn.dairo.cls.dao.dto.ForwardDto

/**
 * 数据转发操作
 */
object ForwardDao {

    /**
     * 添加一条隧道
     */
    fun add(dto: ForwardDto) {
        val updateDate = System.currentTimeMillis()
        val sql =
            "insert into forward(name,port,targetPort,aclState,enableState,updateDate)values(?,?,?,?,?,?)"
        NPSDB.db.exec(
            sql,
            dto.name,
            dto.port,
            dto.targetPort,
            dto.aclState,
            dto.enableState,
            updateDate
        )

        val insertIdSql = "select id from forward where updateDate = ?"
        val id = NPSDB.db.selectSingleOne<Int>(insertIdSql, updateDate)
        dto.id = id
    }

    /**
     * 通过id获取一条数据
     * @param id 隧道id
     * @return 隧道Dto
     */
    fun selectOne(id: Int): ForwardDto? {
        val sql = "select * from forward where id = ?"
        val dto = NPSDB.db.selectOne(ForwardDto::class.java, sql, id)
        return dto
    }

    /**
     * 获取所有数据
     * @return 隧道Dto
     */
    fun selectAll(): List<ForwardDto> {
        val sql = "select * from forward"
        val list = NPSDB.db.selectList(ForwardDto::class.java, sql)
        return list
    }

    /**
     * 获取所有数据
     * @return 隧道Dto
     */
    fun selectActive(): List<ForwardDto> {
        val sql = "select * from forward where enableState = 1"
        val list = NPSDB.db.selectList(ForwardDto::class.java, sql)
        return list
    }



    /**
     * 更新一条数据
     */
    fun update(dto: ForwardDto) {
        val sql =
            "update forward set name = ?,port=?,targetPort=?,enableState=?,aclState=?,remark=?,updateDate=${System.currentTimeMillis()} where id = ? and updateDate=?"
        NPSDB.db.exec(
            sql,
            dto.name,
            dto.port,
            dto.targetPort,
            dto.enableState,
            dto.aclState,
            dto.remark,
            dto.id,
            dto.updateDate
        )
    }

    /**
     * 同步入出网流量
     */
    fun setDataLen(dto: ForwardDto) {
        val sql = "update forward set inDataTotal = ?,outDataTotal=? where id = ?"
        NPSDB.db.exec(sql, dto.inDataTotal, dto.outDataTotal, dto.id)
    }

    /**
     * 通过id删除一条数据
     * @param id 隧道id
     */
    fun delete(id: Int) {
        val sql = "delete from forward where id = ?"
        NPSDB.db.exec(sql, id)
    }

    /**
     * 设置备注信息
     */
    fun setRemark(id: Int, remark: String) {
        val sql =
            "update forward set remark = ? where id = ?"
        NPSDB.db.exec(sql, remark, id)
    }
}

File: dao/ForwardDao/ForwardDao.go
package ForwardDao

				import (
					"DairoNPS/DebugTimer"
				)

import (
	"DairoNPS/dao/dto"
	"DairoNPS/util/DBUtil"
)

func init() {
DebugTimer.Add93()
	ClearError()
}

/**
 * 添加一条隧道
 */
func Add(dto *dto.ForwardDto) {
DebugTimer.Add94()
	sql :=
		"insert into forward(name,port,targetPort,aclState)values(?,?,?,?)"
	id := DBUtil.InsertIgnoreError(
		sql,
		dto.Name,
		dto.Port,
		dto.TargetPort,
		dto.AclState,
	)
	dto.Id = int(id)
}

/**
 * 通过id获取一条数据
 * @param id 隧道id
 * @return 隧道Dto
 */
func SelectOne(id int) *dto.ForwardDto {
DebugTimer.Add95()
	sql := "select * from forward where id = ?"
	return DBUtil.SelectOne[dto.ForwardDto](sql, id)
}

/**
 * 获取所有数据
 * @return 隧道Dto
 */
func SelectAll() []*dto.ForwardDto {
DebugTimer.Add96()
	sql := "select * from forward"
	return DBUtil.SelectList[dto.ForwardDto](sql)
}

/**
 * 获取所有数据
 * @return 隧道Dto
 */
func SelectActive() []*dto.ForwardDto {
DebugTimer.Add97()
	sql := "select * from forward where enableState = 1"
	return DBUtil.SelectList[dto.ForwardDto](sql)
}

// 通过端口查询一条数据
func SelectByPort(port int) *dto.ForwardDto {
DebugTimer.Add98()
	sql := "select * from forward where port = ?"
	return DBUtil.SelectOne[dto.ForwardDto](sql, port)
}

/**
 * 更新一条数据
 */
func Update(dto *dto.ForwardDto) {
DebugTimer.Add99()
	sql :=
		"update forward set name = ?,port=?,targetPort=?,aclState=?,remark=? where id = ?"
	DBUtil.ExecIgnoreError(
		sql,
		dto.Name,
		dto.Port,
		dto.TargetPort,
		dto.AclState,
		dto.Remark,
		dto.Id,
	)
}

// 设置可用状态
func SetEnableState(id int, state int) {
DebugTimer.Add100()
	sql := "update forward set enableState = ? where id = ?"
	DBUtil.ExecIgnoreError(sql, state, id)
}

/**
 * 同步入出网流量
 */
func SetDataSize(id int, inData int64, outData int64) {
DebugTimer.Add101()
	sql := "update forward set inData = ?,outData=? where id = ?"
	DBUtil.ExecIgnoreError(sql, inData, outData, id)
}

/**
 * 通过id删除一条数据
 * @param id 隧道id
 */
func Delete(id int) {
DebugTimer.Add102()
	sql := "delete from forward where id = ?"
	DBUtil.Exec(sql, id)
}

/**
 * 设置备注信息
 */
func SetRemark(id int, remark string) {
DebugTimer.Add103()
	sql := "update forward set remark = ? where id = ?"
	DBUtil.Exec(sql, remark, id)
}

// 设置错误信息
func SetError(id int, error *string) {
DebugTimer.Add104()
	sql := "update forward set error = ? where id = ?"
	DBUtil.ExecIgnoreError(sql, error, id)
}

// 清空错误信息
func ClearError() {
DebugTimer.Add105()
	sql := "update forward set error = null"
	DBUtil.ExecIgnoreError(sql)
}
